pkg/obfuscate/shell_lexer.go

```diff
@@ -0,0 +1,395 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package obfuscate
+
+import (
+	"regexp"
+	"strings"
+)
+
+// ShellTokenKind is a type of shell token.
+type ShellTokenKind int
+
+// ShellTokenKind values
+const (
+	Executable ShellTokenKind = iota
+	Field
+	DoubleQuote
+	SingleQuote
+	Control
+	VariableDefinition
+	WhiteSpace
+	Equal
+	Backticks
+	Dollar
+	ShellVariable
+	Redirection
+	ParentheseOpen
+	ParentheseClose
+)
+
+// Shared regex patterns
+var (
+	IFS         = " \n\r\t"
+	expressions = map[string]*regexp.Regexp{
```

Comment:
do you expect to ever write to this? It might be a bit faster (and I think just as readable) to not have this as a map and just hard code these values individually